spec/controllers/api/v4/drug_stocks_controller_spec.rb

```diff
@@ -0,0 +1,77 @@
+require "rails_helper"
+
+RSpec.describe Api::V4::DrugStocksController, type: :controller do
+  let(:user) { create(:user) }
+  let(:facility_group) { user.facility.facility_group }
+  let(:facility) { create(:facility, facility_group: facility_group) }
+
+  describe "#index" do
+    render_views
+
+    it "returns the latest drug stocks for the user's facility" do
+      month = Date.parse("2021-10-29")
+      end_of_month = month.end_of_month
+      drug_stocks = create_list(:drug_stock, 3, facility: facility, for_end_of_month: month)
+
+      expected_response = {
+        "month" => "2021-10",
+        "facility_id" => facility.id,
+        "drugs" => drug_stocks.map do |stock|
+          {
+            "protocol_drug_id" => stock.protocol_drug_id,
+            "in_stock" => stock.in_stock,
+            "received" => stock.received
+          }
+        end
+      }
+
+      allow(DrugStock).to receive(:latest_for_facilities).with([facility], end_of_month).and_return(drug_stocks)
```

Comment:
Is this mocking needed if the drug stocks have been created in the right month at `facility`?

Reply:
Ah it is indeed not. I will remove.